.github/workflows/testrail-api-tests.yml

```diff
@@ -0,0 +1,65 @@
+name: testrail-api-tests
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Run every day at midnight
+  push:
+    paths:
+      - 'testrail/**' # Run on changes to testrail directory
+  pull_request:
+    paths:
+      - 'testrail/**' # Run on changes to testrail directory
+
+jobs:
+  test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+
+    env:
+      TESTRAIL_HOST: ${{ secrets.TESTRAIL_HOST }}
+      TESTRAIL_USERNAME: ${{ secrets.TESTRAIL_USERNAME }}
+      TESTRAIL_PASSWORD: ${{ secrets.TESTRAIL_PASSWORD }}
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL_TEST_ALERTS_SANDBOX }}
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.11
+
+    - name: Install dependencies
+      run: |
+        pip install -r testrail/requirements.txt
+
+    - name: Run tests
+      run: |
+        python -m unittest discover -s testrail/tests -p '*tests.py' > test_results.txt 2>&1
+
+    - name: Send Slack message on test failure
+      if: ${{ failure() }}
+      id: slack-failure
+      uses: slackapi/slack-github-action@v1.26.0
+      with:
+        channel-id: ${{ secrets.SLACK_MOBILE_ALERTS_SANDBOX_CHANNEL }}
+        payload: |
+          {
+            "type": "mrkdwn",
+            "text": "TestRail API tests failed! :tada:\n\n*Workflow:* ${{ github.workflow }}\n*Job:* ${{ github.job }}\n\n*Status:* ${{ job.status }}\n*Test results:* Failed\n\nLink to results: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|View results>"
+          }
+        
+
+    - name: Send Slack message on test success
+      if: ${{ success() }}
+      id: slack-success
+      uses: slackapi/slack-github-action@v1.26.0
+      with:
+        channel-id: ${{ secrets.SLACK_MOBILE_ALERTS_SANDBOX_CHANNEL }}
+        payload: |
```

Comment:
Same here. Can probably specify a success template vs a failure template payload file too if needed.

Reply:
yeah, forgot to add one. figure we can add a template when we start rolling them out in mobile-ci-messaging stuffs?